libraries/AP_NavEKF2/AP_NavEKF2_MagFusion.cpp

```diff
@@ -388,6 +389,17 @@ void NavEKF2_core::FuseMagnetometer()
         MagPred[1] = DCM[1][0]*magN + DCM[1][1]*magE  + DCM[1][2]*magD + magYbias;
         MagPred[2] = DCM[2][0]*magN + DCM[2][1]*magE  + DCM[2][2]*magD + magZbias;
 
+        if (pending_z_mag_correction && frontend->_mag_learn_limit > 0) {
+            pending_z_mag_correction = false;
+            float correction = magDataDelayed.mag[2] - MagPred[2];
+            const float correction_limit_ga = frontend->_mag_learn_limit*0.001*3;
+            correction = constrain_float(correction,
+                                         -correction_limit_ga, correction_limit_ga);
+            stateStruct.body_magfield[2] += correction;
```

Comment:
2 needs to be a const.

Reply:
no, it doesn't. Please don't ask for changes in code style away from how it is done in the upstream ardupilot code. I am working to improve the code style there, but diverging within matternet would be a very bad idea